### PR TITLE
Shoulder Sling attachment

### PR DESCRIPTION
**Pull request by Gristlebee**
## About The Pull Request

Adds the shoulder sling attachment to cargo for 500 credits.

The shoulder sling can be attached any long arm gun with a weight class of bulky or up. The sling allows for the gun to be carried in your suit storage slot without an armor holster.

Adds support for attachment overlays to render under the attached gun.

Shoulder sling sprites by Moffball.

image

## Why It's Good For The Game

This gives players more options for fashion and accessorizing. You're no longer stuck with wearing a boring armor vest to hold your bulky weapons, and the other option of carrying your gun everywhere in your hands makes you look like a maniac.  Balance wise this should be fine, since you can buy armor from the outpost anyways, which provides the same benefit with it's holster while being objectively better because well. It's armored.

## Changelog

:cl:
add: Shoulder slings to outpost cargo for 500 credits.
/:cl:

### CODE REVIEW

**Comment by DIB-DOG** [comment]
Suggestion of perhaps renaming it to “shoulder sling” or “weapon sling”? I think it would be more fitting given what is being described/depicted.

**Comment by Gristlebee** [comment]
> Suggestion of perhaps renaming it to “shoulder sling” or “weapon sling”? I think it would be more fitting given what is being described/depicted.

Yeah, that sounds better.

File: code/game/objects/items/attachments/shoulder_sling.dm [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/obj/item/attachment/sling
+	name = "shoulder sling"
+	desc = "A leather shoulder sling for longarms to allow for easy carrying on the shoulder without the need for traditional armor holsters."
+	icon_state = "sling"
```
Comment (thgvr) [inline code/game/objects/items/attachments/shoulder_sling.dm:4]: You didn't rename the icon state in the DMI

File: code/modules/cargo/packs/gun.dm [hunk 2]
```diff
@@ -644,3 +644,10 @@
 	cost = 250
 	contains = list(/obj/item/attachment/silencer)
 	crate_name = "suppressor crate"
+
+/datum/supply_pack/gun/attachment/sling
+	name = "Shoulder Sling Crate"
+	desc = "Contains a single shoulder sling to be mounted on a firearm for easy carrying without armor holsters. Only compatible with longarms."
+	cost = 500
```
Comment (thgvr) [inline code/modules/cargo/packs/gun.dm]: pretty expensive. 250?

File: code/game/objects/items/attachments/shoulder_sling.dm (hunk 1 above)
Comment (Gristlebee) [reply code/game/objects/items/attachments/shoulder_sling.dm:4]: Fixed

File: code/modules/cargo/packs/gun.dm (hunk 2 above)
Comment (Gristlebee) [reply code/modules/cargo/packs/gun.dm]: Sure, that sounds good.

**Pull request merged**